community/primitive-collections/src/main/java/org/neo4j/collection/primitive/PrimitiveLongStack.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2002-2016 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.collection.primitive;
+
+import static java.util.Arrays.copyOf;
+
+/**
+ * Like a {@code Stack<Long>} but for primitive longs. Virtually GC free in that it has an {@code long[]}
+ * and merely moves a cursor where to {@link #push(long)} and {@link #poll()} values to and from.
+ * If many items goes in the stack the {@code long[]} will grow to accomodate all of them, but not shrink again.
+ */
+public class PrimitiveLongStack implements PrimitiveLongCollection
```

Comment:
A test for this guy would be nice, just to make sure there isn't any dumb mistakes hiding in here.